Tests/Functional/Controller/FrontEndEditorControllerTest.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+
+namespace TTN\Tea\Tests\Functional\Controller;
+
+use PHPUnit\Framework\Attributes\CoversClass;
+use PHPUnit\Framework\Attributes\Test;
+use TTN\Tea\Controller\FrontEndEditorController;
+use TYPO3\TestingFramework\Core\Functional\Framework\Frontend\InternalRequest;
+use TYPO3\TestingFramework\Core\Functional\Framework\Frontend\InternalRequestContext;
+
+#[CoversClass(FrontEndEditorController::class)]
+final class FrontEndEditorControllerTest extends AbstractFrontendControllerTestCase
+{
+    private const PAGE_UID = 1;
+    private const FRONTEND_USER_UID = 1;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->importCSVDataSet(__DIR__ . '/Fixtures/Database/ContentElementTeaFrontEndEditor.csv');
+        $this->importCSVDataSet(__DIR__ . '/Fixtures/Database/FrontendUsers.csv');
+    }
+
+    #[Test]
+    public function indexActionForNoLoggedInUser(): void
+    {
+        $request = (new InternalRequest())->withPageId(self::PAGE_UID);
+
+        $html = (string)$this->executeFrontendSubRequest($request)->getBody();
+
+        self::assertStringContainsString('Please configure this plugin to be only visible if a website user is logged in.', $html);
+    }
+
+    #[Test]
+    public function indexActionForLoggedInUserRendersTeasOwnedByTheLoggedInUser(): void
```

Comment:
Please let's split this test into three tests:

- `indexActionForLoggedInUserRendersTeaOwnedByTheLoggedInUser`
- `indexActionForLoggedInUserDoesNotRenderTeaOwnedByOtherUser`
- `indexActionForLoggedInUserDoesNotRenderTeaWithoutOwner`